terraform/ec2/common/linux/main.tf

```diff
@@ -55,6 +55,14 @@ resource "aws_instance" "cwagent" {
   vpc_security_group_ids               = [module.basic_components.security_group]
   associate_public_ip_address          = true
   instance_initiated_shutdown_behavior = "terminate"
+  # Provide a user_data script to disable SSH password authentication
+  user_data = <<-EOT
+    #!/bin/bash
+    # Disable password authentication for SSH
+    sed -i 's/^#*PasswordAuthentication.*/PasswordAuthentication no/' /etc/ssh/sshd_config
```

Comment:
let's make sure we're adding the appropriate ChallengeResponseAuthentication as well

Reply:
Good call out!